src/main/java/org/fogbowcloud/manager/core/services/OrdersService.java

```diff
@@ -0,0 +1,58 @@
+package org.fogbowcloud.manager.core.services;
+
+import org.fogbowcloud.manager.core.controllers.ApplicationController;
+import org.fogbowcloud.manager.core.datastructures.SharedOrderHolders;
+import org.fogbowcloud.manager.core.exceptions.OrdersServiceException;
+import org.fogbowcloud.manager.core.models.orders.Order;
+import org.fogbowcloud.manager.core.models.orders.OrderState;
+import org.fogbowcloud.manager.core.models.token.Token;
+import org.fogbowcloud.manager.core.plugins.identity.exceptions.UnauthorizedException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+import java.util.Map;
+
+public class OrdersService {
+
+    // TODO: usar log ao longo do código
+    private static final Logger LOGGER = LoggerFactory.getLogger(OrdersService.class);
+
+    private SharedOrderHolders ordersHolder;
+
+    public OrdersService() {
+        ordersHolder = SharedOrderHolders.getInstance();
+    }
+
+    public void createOrder(Order order, Token federatedToken, Token localToken) throws OrdersServiceException, UnauthorizedException {
+        // TODO para os headers
+        if (order == null) {
+            // TODO create msg
+            String msg = "";
+            throw new OrdersServiceException(msg);
+        }
+        setOrderTokens(order, localToken, federatedToken);
+        order.setOrderState(OrderState.OPEN);
+        addOrderInActiveOrdersMap(order);
+    }
+
+    private void setOrderTokens(Order order, Token federatedToken, Token localToken) {
```

Comment:
I think this method a little bit odd. Not against, though.  Let's see how the code evolve until it reaches the develop branch

Reply:
You are right. I think it was an overkill.